### PR TITLE
Add convolution 2d custom layer

### PR DESCRIPTION
**Pull request by gmuraru**
## Description
Add the 2d Conv layer.

## How has this been tested?
- Added test for inference and reconstruction

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: src/sympc/module/nn/conv.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        module = torch.nn.Conv2d(
+            in_channels=in_channels,
+            out_channels=out_channels,
+            kernel_size=kernel_size_w,
+            bias=bias,
+        )
+        return module
+
+    @staticmethod
+    def eq_close(
```
Comment (LaRiffle) [inline src/sympc/module/nn/conv.py:100]: Maybe this could be a method of a parent shared class between Linear and Conv2d -> a nice GFI?

File: src/sympc/tensor/mpc_tensor.py [hunk 2]
```diff
@@ -426,7 +426,7 @@ def conv2d(
         convolution = self.__apply_op(weight, "conv2d", kwargs_=kwargs)
 
         if bias:
-            return convolution + bias
+            return convolution + bias.unsqueeze(1).unsqueeze(1)
```
Comment (LaRiffle) [inline src/sympc/tensor/mpc_tensor.py:429]: Are you sure the double unsqueeze is needed?

File: src/sympc/module/nn/conv.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        "kernel_size",
+        "stride",
+        "padding",
+        "dilation",
+        "groups",
+    ]
+
+    in_channels: int
+    out_channels: int
+    kernel_size: Tuple[int, ...]
```
Comment (LaRiffle) [inline src/sympc/module/nn/conv.py:29]: Why are these defined but not set as `self.`, contrary to the next `stride`, `padding`... ?

File: src/sympc/tensor/mpc_tensor.py (hunk 2 above)
Comment (gmuraru) [reply src/sympc/tensor/mpc_tensor.py:429]: There was a problem here because the bias was not broadcasted accordingly :(
Bias was having the shape [10] and I think the convolution was something like [10, X, X] where X was a number.

File: src/sympc/module/nn/conv.py (hunk 3 above)
Comment (gmuraru) [reply src/sympc/module/nn/conv.py:29]: Hmm...I have seen that you could specify the type of the attributes, without them being bound to an instance like this (even if some of them are not used).
Should I remove them? (I do not have any strong opinion regarding this, I set those because I have seen them defined like this [in torch](https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/conv.py#L56))

Comment (LaRiffle) [reply src/sympc/module/nn/conv.py:31]: I don't know to be honest! :)

File: src/sympc/tensor/mpc_tensor.py (hunk 2 above)
Comment (LaRiffle) [reply src/sympc/tensor/mpc_tensor.py:429]: Hmmm that's strange! But ok let's keep it as such for now

**Review by LaRiffle — APPROVED** [review]
Looks good to me!

**Review by arturomf94 — APPROVED** [review]
